src/components/AddressSearch.js

```diff
@@ -36,51 +35,44 @@ const defaultProps = {
 
     const saveLocationDetails = (details) => {
         const addressComponents = details.address_components;
-        if (isAddressValidForVBA(addressComponents)) {
+        if (addressComponents) {
             // Gather the values from the Google details
-            const streetNumber = getAddressComponent(addressComponents, 'street_number', 'long_name');
-            const streetName = getAddressComponent(addressComponents, 'route', 'long_name');
+            const streetNumber = getAddressComponent(addressComponents, 'street_number', 'long_name') || '';
+            const streetName = getAddressComponent(addressComponents, 'route', 'long_name') || '';
+
+            const streetNumberAndName = `${streetNumber} ${streetName}`.trim();
+
             let city = getAddressComponent(addressComponents, 'locality', 'long_name');
             if (!city) {
                 city = getAddressComponent(addressComponents, 'sublocality', 'long_name');
                 Log.hmmm('[AddressSearch] Replacing missing locality with sublocality: ', {address: details.formatted_address, sublocality: city});
             }
-            const state = getAddressComponent(addressComponents, 'administrative_area_level_1', 'short_name');
             const zipCode = getAddressComponent(addressComponents, 'postal_code', 'long_name');
+            const state = getAddressComponent(addressComponents, 'administrative_area_level_1', 'short_name');
 
             // Trigger text change events for each of the individual fields being saved on the server
-            props.onChangeText('addressStreet', `${streetNumber} ${streetName}`);
-            props.onChangeText('addressCity', city);
-            props.onChangeText('addressState', state);
-            props.onChangeText('addressZipCode', zipCode);
-        } else {
-            // Clear the values associated to the address, so our validations catch the problem
-            Log.hmmm('[AddressSearch] Search result failed validation: ', {
-                address: details.formatted_address,
-                address_components: addressComponents,
-                place_id: details.place_id,
-            });
-            props.onChangeText('addressStreet', null);
-            props.onChangeText('addressCity', null);
-            props.onChangeText('addressState', null);
-            props.onChangeText('addressZipCode', null);
+            if (streetNumberAndName.length > props.value.length) {
+                // We only replace what the user typed in the address input if we think ours is more complete
+                props.onChangeText('addressStreet', streetNumberAndName);
+            }
+            if (city) {
+                props.onChangeText('addressCity', city);
+            }
+            if (state) {
+                props.onChangeText('addressState', state);
+            }
+            if (zipCode) {
+                props.onChangeText('addressZipCode', zipCode);
```

Comment:
I know this wasn't added in this PR, but dislike this design. The `onChangeText` can get called multiple times with different key/values and it's really unusual. More typical would be a single text input that returns a text value for one field only.

Does it feel like we should be passing back an object or something here to just one callback and then letting the parent component determine what "fields" should be using the data?

I think then we would not need to worry about if these values exist just do

```js
props.onLocationSelected({city, state, zipCode, streetNumberAndName});
```

Reply:
Totally agree with this